File: week1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    //ask for height
    int height;
    
    do{
          height = get_int("Height = "); 
    }
    while (height > 8 || height < 1);  

    //building the pryramid
    
    for (int x = 0; x < height; x++)                // defining rows (x-axis)
    {

        
        for (int z = 0; z < (height - x - 1); z++)  // building dots/space, relationship found: the spaces needed =  height - x -1
        {
            printf(" ");
        }
      
        for (int y = 0; y < x+1; y++)               //building for #
        {
           printf ("#");  
        }
    
    printf("\n");                                   //print out new lines
       
    }
}

File: week6/mario.py
from cs50 import get_int

# Get height by using a while loop (height must be more than 0 or <=8)
while True:
    height = get_int("Height:")
    if height > 0 and height <= 8:
        break

for i in range(height):
   
    # From C code: building dots/space, relationship found: the spaces needed =  height - x -1
    for j in range(0, height - i - 1, +1):
        print(" ", end="")
    
    # From C code
    for y in range(0, i + 1, +1):
        print("#", end="")
    
    # print new line after finishing a line of " " and  "#"
    print()
